_posts/2023-08-23-even-friendlier-sql.md

```diff
@@ -0,0 +1,596 @@
+     ON COLUMNS(* EXCLUDE item)
+     INTO
+          NAME year
+          VALUE count;
+```
+
+|       item       | year | count |
+|:---|:---|:---|
+| phasers          | 2155 | 1035  |
+| phasers          | 2156 | 25039 |
+| phasers          | 2157 | 95000 |
+| photon torpedoes | 2155 | 255   |
+| photon torpedoes | 2156 | 17899 |
+| photon torpedoes | 2157 | 87492 |
+
+More examples are included as a part of our [DuckDB 0.8.0 announcement post](https://duckdb.org/2023/05/17/announcing-duckdb-080.html#new-sql-features), and the [`PIVOT`](https://duckdb.org/docs/sql/statements/pivot) and [`UNPIVOT`](https://duckdb.org/docs/sql/statements/unpivot) documentation pages highlight more complex queries. 
+
+Stay tuned for a future post to cover what is happening behind the scenes! 
+
+## List lambda functions
+
+List lambdas allow for operations to be applied to each item in a list. These do not need to be pre-defined - they are created on the fly within the query. 
+
+In this example, a lambda function is used in combination with the `list_transform` function to shorten each official ship name. 
+
+```sql
+SELECT 
+     (['Enterprise NCC-1701', 'Voyager NCC-74656', 'Discovery NCC-1031'])
+          .list_transform(x -> x.string_split(' ')[1]) AS short_name;
+```
+
+|            ship_name             |
+|:---|
+| [Enterprise, Voyager, Discovery] |
+
+Lambdas can also be used to filter down the items in a list. The lambda returns a list of booleans, which is used by the `list_filter` function to select specific items. The `contains` function is using the [function chaining](#function-chaining) described earlier.
+
+```sql
+SELECT 
+     (['Enterprise NCC-1701', 'Voyager NCC-74656', 'Discovery NCC-1031'])
+          .list_filter(x -> x.contains('1701')) AS the_original;
+```
+
+|     the_original      |
+|:---|
+| [Enterprise NCC-1701] |
+
+## List comprehensions
+
+What if there was a simple syntax to both modify and filter a list? DuckDB takes inspiration from Python’s approach to list comprehensions to dramatically simplify the above examples. List comprehensions are syntactic sugar - these queries are rewritten into lambda expressions behind the scenes!
+
+Within brackets, first specify the transformation that is desired, then indicate which list should be iterated over, and finally include the filter criteria. 
+
+```sql
+SELECT 
+     [x.string_split(' ')[1] 
+     FOR x IN ['Enterprise NCC-1701', 'Voyager NCC-74656', 'Discovery NCC-1031'] 
+     IF x.contains('1701')] AS ready_to_boldly_go;
+```
+
+| ready_to_boldly_go |
+|:---|
+| [Enterprise]       |
+
+## Exploding Struct.*
+
+A struct in DuckDB is a set of key/value pairs. Behind the scenes, a struct is stored with a separate column for each key. As a result, it is computationally easy to explode a struct into separate columns, and now it is also syntactically simple as well! This is another example of allowing SQL to handle dynamic column names.
+
+```sql
+WITH damage_report as (
```

Comment:
`as` -> `AS`?
Sorry... :) Just pushed c57c5ad5dfaf1fd4d828f604ed1e430d41389d14 today, which fixed this across the documentation.